src/gpio/convert.rs

```diff
@@ -0,0 +1,477 @@
+        Otype,
+    > From<Pin<P, N, MODE>> for Pin<P, N, Alternate<A, Otype>>
+where
+    Alternate<A, Otype>: PinMode,
+    Self: marker::IntoAf<A>,
+{
+    #[inline(always)]
+    fn from(f: Pin<P, N, MODE>) -> Self {
+        f.into_mode()
+    }
+}
+
+impl<const P: char, const N: u8, const A: u8, const B: u8>
+    From<Pin<P, N, Alternate<B, PushPull>>>
+    for Pin<P, N, Alternate<A, OpenDrain>>
+where
+    Self: marker::IntoAf<A>,
+{
+    #[inline(always)]
+    fn from(f: Pin<P, N, Alternate<B, PushPull>>) -> Self {
+        f.into_mode()
+    }
+}
+
+impl<const P: char, const N: u8, Otype> From<Pin<P, N, Output<Otype>>>
+    for Pin<P, N, Input>
+where
+    Output<Otype>: PinMode,
+{
+    #[inline(always)]
+    fn from(f: Pin<P, N, Output<Otype>>) -> Self {
+        f.into_mode()
+    }
+}
+
+impl<const P: char, const N: u8> From<Pin<P, N, Analog>> for Pin<P, N, Input> {
+    #[inline(always)]
+    fn from(f: Pin<P, N, Analog>) -> Self {
+        f.into_mode()
+    }
+}
+
+impl<const P: char, const N: u8, const A: u8, Otype, MODE>
+    From<Pin<P, N, Alternate<A, Otype>>> for Pin<P, N, MODE>
+where
+    Alternate<A, Otype>: PinMode,
+    MODE: PinMode + marker::NotAlt,
+{
+    #[inline(always)]
+    fn from(f: Pin<P, N, Alternate<A, Otype>>) -> Self {
+        f.into_mode()
+    }
+}
+
+impl<const P: char, const N: u8, Otype> From<Pin<P, N, Input>>
+    for Pin<P, N, Output<Otype>>
+where
+    Output<Otype>: PinMode,
+{
+    #[inline(always)]
+    fn from(f: Pin<P, N, Input>) -> Self {
+        f.into_mode()
+    }
+}
+
+impl<const P: char, const N: u8, Otype> From<Pin<P, N, Analog>>
+    for Pin<P, N, Output<Otype>>
+where
+    Output<Otype>: PinMode,
+{
+    #[inline(always)]
+    fn from(f: Pin<P, N, Analog>) -> Self {
+        f.into_mode()
+    }
+}
+
+impl<const P: char, const N: u8> From<Pin<P, N, Output<PushPull>>>
+    for Pin<P, N, Output<OpenDrain>>
+{
+    #[inline(always)]
+    fn from(f: Pin<P, N, Output<PushPull>>) -> Self {
+        f.into_mode()
+    }
+}
+
+impl<const P: char, const N: u8> From<Pin<P, N, Output<OpenDrain>>>
+    for Pin<P, N, Output<PushPull>>
+{
+    #[inline(always)]
+    fn from(f: Pin<P, N, Output<OpenDrain>>) -> Self {
+        f.into_mode()
+    }
+}
+
+impl<const P: char, const N: u8> From<Pin<P, N, Input>> for Pin<P, N, Analog> {
+    #[inline(always)]
+    fn from(f: Pin<P, N, Input>) -> Self {
+        f.into_mode()
+    }
+}
+
+impl<const P: char, const N: u8, Otype> From<Pin<P, N, Output<Otype>>>
+    for Pin<P, N, Analog>
+where
+    Output<Otype>: PinMode,
+{
+    #[inline(always)]
+    fn from(f: Pin<P, N, Output<Otype>>) -> Self {
+        f.into_mode()
+    }
+}
+
+macro_rules! af {
+    ($($into_alternate_af:ident: $A:literal;)+) => {
+        $(
+            #[doc="Configures the pin to operate in alternate function "]
+            #[doc=stringify!($A)]
+            #[doc=" mode"]
+            #[deprecated(since = "0.12.0", note = "Use the .into_alternate() method instead")]
```

Comment:
I don't think there should be deprecated functions introduced here ;)

Reply:
Oops. Copy paste error. Thanks for catching!